embassy-stm32/src/eth/generic_smi.rs

```diff
@@ -38,7 +42,22 @@ mod phy_consts {
 use self::phy_consts::*;
 
 /// Generic SMI Ethernet PHY
-pub struct GenericSMI;
+pub struct GenericSMI {
+    #[cfg(feature = "time")]
+    poll_interval: Duration,
+}
+
+impl GenericSMI {
+    #[cfg(feature = "time")]
+    pub fn new(poll_interval: Duration) -> Self {
```

Comment:
we should try to make Cargo features additive, so working code doesn't break if you enable `time`.

 Instead of adding/removing the param, could you make `new()` default to e.g. 500ms and then add a `.set_poll_interval(Duration)` fn?